src/js/profiling/hermes.ts

```diff
@@ -0,0 +1,91 @@
+import { NATIVE } from '../wrapper';
+import { convertToSentryProfile } from './convertHermesProfile';
+import type { RawThreadCpuProfile } from './types';
+
+export type StackFrameId = number;
+export type MicrosecondsSinceBoot = string;
+
+export interface TraceEvent {
+  name: string;
+  ph: string;
+  cat: string;
+  pid: number;
+  ts: MicrosecondsSinceBoot;
+  tid: string;
+  args: {
+    name?: string;
+  };
+}
+
+export interface Sample {
+  cpu: string;
+  name: string;
+  ts: MicrosecondsSinceBoot;
+  pid: number;
+  tid: string;
+  weight: string;
+  sf: StackFrameId;
+}
+
+export interface StackFrame {
+  line?: string;
+  column?: string;
+  funcLine?: string;
+  funcColumn?: string;
+  name: string;
+  category: string;
+  parent?: number;
+}
+
+export interface Profile {
+  traceEvents: TraceEvent[];
+  samples: Sample[];
+  stackFrames: Record<string, StackFrame>;
+}
+
+export interface HermesStackFrameNameParsed {
+  function?: string;
+  fileName?: string;
+}
+
+const FILE_NAME_REGEX = /([a-zA-Z0-9_-]+\.[a-zA-Z0-9_-]+)(?::(\d+:\d+))?\)/;
+/**
+ * Hermes Profile Stack Frame Name contains function name and file path.
+ *
+ * `foo(/path/to/file.js:1:2)` -> `{ function: 'foo', fileName: 'file.js' }`
+ */
+export function parseHermesStackFrameName(name: string): HermesStackFrameNameParsed {
```

Comment:
I have a hunch this will be perf sensitive. Have you done any benchmarking on this?

Reply:
I didn't do benchmarking on this, I will take a look on it.

Reply:
Yeah, I wouldn't worry too much about it, def a topic for a different PR. Just something to be mindful about as we might be parsing a lot of frames here

Reply:
The original code with regex takes ~10ms on my simulator. The new code without the regex takes ~4ms. Thanks for pointing this out.